src/app/app.component.ts

```diff
@@ -11,7 +14,65 @@ import { DEPARTURE_AIRCRAFT_STATUS } from 'src/assets/mock-data/mock-aircraft-st
 })
 export class AppComponent {
 
-  ngOnInit() { }
+  private aircraftStatusUrl = "/assets/mock-data/mockAircraftStatus.json";
+  private tripsUrl = "assets/mock-data/mockTrips.json";
+  isDataAvailable = false;
+
+  departureAircraftStatus: AircraftStatus | undefined;
+  currentBoardingTime: any;
+  estimatedBoardingTime: any;
+  boardingTimeString: any;
+  boardingTimeDifference: any;
+  boardingTimeEarlyOrLate: any;
+  boardingTimeMinOrMins: any;
+  boardingLastRefresh: any;
+  boardingGate: any;
+
+  trip: any;
+
+  departureMonth: any;
+  departureDateTimeString: any;
+  arrivalMonth: any;
+  arrivalDateTimeString: any;
+
+  constructor(private http: HttpClient) { }
+
+  ngOnInit() {
+    this.getJSON(this.tripsUrl).pipe(
+      switchMap((data) => {
+        return forkJoin({
+          trips: of(data),
+          aircraftStatus: this.getJSON(this.aircraftStatusUrl)
+        });
+      })
+    ).subscribe(({ trips, aircraftStatus }) => {
+      this.trip = <Trip>trips.currentTrip;
+
+      this.departureAircraftStatus = <AircraftStatus>aircraftStatus.departureAircraftStatus;
+      this.currentBoardingTime = new Date(this.departureAircraftStatus.currentBoardingTime);
+      this.estimatedBoardingTime = new Date(this.departureAircraftStatus.estimatedBoardingTime);
+      this.boardingTimeString = `${this.currentBoardingTime.getHours()}:${addZeroToTime(this.currentBoardingTime.getMinutes())}`;
+      this.boardingTimeDifference = this.getTimeDifference(this.currentBoardingTime, this.estimatedBoardingTime);
+      this.boardingTimeEarlyOrLate = this.earlyOrLate(this.currentBoardingTime, this.estimatedBoardingTime);
+      this.boardingTimeMinOrMins = this.minOrMins(this.currentBoardingTime, this.estimatedBoardingTime)
+      this.boardingLastRefresh = Math.floor((Math.abs(new Date().getTime() - new Date().getTime()) / 1000) / 60);
+      this.boardingGate = this.departureAircraftStatus.gate;
+
+      this.departureMonth = MONTHS[new Date(this.departureAircraftStatus.departureTime).getMonth()];
+      this.departureDateTimeString = getDateAsDDTTTT(new Date(this.departureAircraftStatus.departureTime));
+      this.arrivalMonth = MONTHS[new Date(this.departureAircraftStatus.arrivalTime).getMonth()];
+      this.arrivalDateTimeString = getDateAsDDTTTT(new Date(this.departureAircraftStatus.arrivalTime));
+
+      this.isDataAvailable = true;
+    });
+    this.getJSON(this.aircraftStatusUrl).subscribe((data) => {
+
+    })
```

Comment:
What does this subscription do?

Reply:
I forgot to remove it! Thank you for pointing it out!